bootloader.c

```diff
@@ -23,13 +23,19 @@
 #include <stdio.h>
 #include <string.h>
 
+#if PICO_PLATFORM == rp2350-arm-s
+#include <RP2350.h>
+#else
 #include <RP2040.h>
+#endif
```

Comment:
You can't do such comparison on preprocesor macro. Even if `PICO_PLATFORM` is equal to `"rp2040"`, the `#if PICO_PLATFORM == rp2350-arm-s` will be evaluated to true and the build will fail.

You should use some other preprocesor constants or create compile-time definitions like `PICO_PLATFORM_RP2040` in CMakeLists.txt file.